Improve `fleetctl` CLI docs page (#16877)
fleetdm/fleet: docs/Using Fleet/fleetctl-CLI.md

# fleetctl CLI

> fleetctl (pronounced "Fleet control") is a CLI tool for managing Fleet from the command line. Fleetctl enables a GitOps workflow with Fleet.

Reviewer: ```suggestion
fleetctl (pronounced "Fleet control") is a CLI tool for managing Fleet from the command line. fleetctl enables a GitOps workflow with Fleet.
```

fleetctl also provides a quick way to work with all the data exposed by Fleet without having to use the Fleet UI or work directly with the Fleet API.

<div purpose="embedded-content">
   <iframe src="https://www.youtube.com/embed/ERbknt6w8eg" allowfullscreen></iframe>
</div>

## Installing fleetctl

> Install the fleetctl with npm or download binary from [GitHub](https://github.com/fleetdm/fleet/releases).

Reviewer: ```suggestion
Install fleetctl with npm or download the binary from [GitHub](https://github.com/fleetdm/fleet/releases).
```

   --context value   Name of fleetctl config context to use (default: "default") [$CONTEXT]
   --debug           Enable debug http request logging (default: false) [$DEBUG]
   --help, -h        show help (default: false)
```

## Authentication

This section walks you through authentication, assuming you already have a running Fleet instance. To learn how to set up new Fleet instance, check out the [Deploy](https://fleetdm.com/docs/deploy/introduction) section or [Building Fleet locally](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/Building-Fleet.md) docs. 

### Login

To log in to your Fleet instance, run following commands:

1. Set the Fleet instance address

```sh
> fleetctl config set --address https://fleet.example.com
[+] Set the address config key to "https://fleet.example.com" in the "default" context
```

> 2. Login with your credentials

Reviewer: ```suggestion
2. Log in with your credentials
```

```sh
> fleetctl login
Log in using the standard Fleet credentials.
Email: mike@arpaia.co
Password:
[+] Fleet login successful and context configured!
```

> Once your local context is configured, you can use the `fleetctl` normally.

Reviewer: ```suggestion
Once your local context is configured, you can use `fleetctl` normally.
```

> ### Login with SAML (SSO) authentication

Reviewer: ```suggestion
### Log in with SAML (SSO) authentication
```

    token: your_token_here
```

The token can also be set with `fleetctl config set --token`, but this may leak the token into a user's shell history.

## Using fleetctl with an API-only user

> When running automated workflows using the Fleet API, we recommend an API-only user's API key rather than the API key of a regular user. Regular user's API key expires frequently for security purposes, requiring routine updates. Meanwhile, an API-only user's key does not expire.   

Reviewer: ```suggestion
When running automated workflows using the Fleet API, we recommend an API-only user's API key rather than the API key of a regular user. A regular user's API key expires frequently for security purposes, requiring routine updates. Meanwhile, an API-only user's key does not expire.   
```

An API-only user does not have access to the Fleet UI. Instead, it's only purpose is to interact with the API programmatically or from fleetctl.

### Create API-only user

To create your new API-only user, use `fleetctl user create`:

```sh
fleetctl user create --name "API User" --email api@example.com --password temp@pass123 --api-only
```

> To use fleetctl with an API-only user, you will need to login via `fleetctl`.  Learn how to login in [section above](https://#authentication).

Reviewer: ```suggestion
To use fleetctl with an API-only user, you will need to log in via `fleetctl`.  See [authentication](https://#authentication) above for details.
```

#### Permissions

An API-only user can be given the same permissions as a regular user. The default access level is **Observer**. You can specify what level of access the new user should have using the `--global-role` flag:

```sh
fleetctl user create --name "API User" --email api@example.com --password temp@pass123 --api-only --global-role admin
```

On Fleet Premium, use the `--team <team_id>:<role>` to create an API-only user on a team:

```sh
fleetctl user create --name "API User" --email api@example.com --password temp@pass123 --api-only --team 4: gitops
```

#### Changing permissions

To change roles of a current user, log into the Fleet UI as an admin and navigate to **Settings > Users**.
> Suggestion: To disable/enable a user's access to the UI (converting a regular user to an API-only user or vice versa), create a new user.

### Get API token for API-only user

> To get the API key of an API-only user, you need to call the [Login API](https://fleetdm.com/docs/rest-api/rest-api#log-in) with the credentials supplied during user creation.

Reviewer: ```suggestion
To get the API key of an API-only user, you need to call the [login API](https://fleetdm.com/docs/rest-api/rest-api#log-in) with the credentials supplied during user creation.
```
